### PR TITLE
Use EditorConfig checker on known file types

### PR DESCRIPTION
**Pull request by colinrotherham**
Lots of code editors and IDEs support [.editorconfig](https://github.com/alphagov/govuk-frontend/blob/main/.editorconfig) by default but we had a problem:

1. Didn't check `.yaml` files added via [The GDS Way: **Code style guides**](https://gds-way.cloudapps.digital/manuals/programming-languages.html)
2. Didn't check `.cjs`, `.mjs` that JavaScript now supports
3. Lots of little fixes needed (see PR https://github.com/alphagov/govuk-frontend/pull/3201)

I've included [EditorConfig ~ESLint plugin~ checker](https://editorconfig-checker.github.io/) to do these checks on PRs too

### CODE REVIEW

**Comment by romaricpascal** [comment]
Cheers for tidying the `.editorconfig`. 😊  I'm not sure checking it's applied should be done via ESLint, though, as that limits it to JS files. Maybe we could bring in [editorconfig-checker](https://editorconfig-checker.github.io/) to have a look at the entire codebase?

**Comment by colinrotherham** [comment]
> Cheers for tidying the `.editorconfig`. 😊 I'm not sure checking it's applied should be done via ESLint, though, as that limits it to JS files. Maybe we could bring in [editorconfig-checker](https://editorconfig-checker.github.io/) to have a look at the entire codebase?

What a great idea 😅

It found lots of little fixes so I've moved them to:

* https://github.com/alphagov/govuk-frontend/pull/3201

**Review by romaricpascal — APPROVED** [review]
Brill! 🙌🏻 Wondering if a run on commit is worthwhile, but I didn't see any "autofix" option 🤔 Happy to look at it in a further PR.

Approved with just a little suggestion on the Nunjucks doc bloc 😄

File: src/govuk/macros/i18n.njk [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
+
+  {@link https://cldr.unicode.org/index/cldr-spec/plural-rules}
+
+  Helps reduce the boilerplate in component templates as they're quite verbose
+
+  @private
+  @param {string} translationKey - The kebab-cased name of the translation key
+  @param {object} pluralForms - An object associating translation messages to
+    the plural form they correspond to
+#}
```
[the remark this answers is not in the document]
Comment (colinrotherham) [reply src/govuk/macros/i18n.njk:12]: @romaricpascal Shall I do the same with these ones?

We had a few multi-line comments that were wrapped to 80 characters, and some that now do

[**editorconfig-checks**/app/views/full-page-examples/cookie-banner-server-side/index.njk#L188](https://github.com/alphagov/govuk-frontend/blob/editorconfig-checks/app/views/full-page-examples/cookie-banner-server-side/index.njk#L188)
```njk
{# As the cookie banner component does not currently include JavaScript, we
  have to take care of the 'Hide' button to make progressive enhancement in
  the example functional #}
```

File: package/govuk/objects/_template.scss [hunk 2]
```diff
@@ -11,8 +11,8 @@
     // Prevent automatic text sizing, as we already cater for small devices and
     // would like the browser to stay on 100% text zoom by default.
     -webkit-text-size-adjust: 100%;
-       -moz-text-size-adjust: 100%;
-            text-size-adjust: 100%;
+    -moz-text-size-adjust: 100%;
+    text-size-adjust: 100%;
```
Comment (36degrees) [inline package/govuk/objects/_template.scss:15]: Feels like Prettier is fighting Autoprefixer here? Does it make senes to run Prettier on our `package`, or should we exclude it?

Comment (colinrotherham) [reply package/govuk/objects/_template.scss:15]: Ah nice spot, think this is just a hangover from before it was ignored
https://github.com/alphagov/govuk-frontend/pull/3201/files#diff-b640b344ee7f3f03d2a443795a5d0708ef50e2e6e34214109ab2aad13ad6ba98

I'll remove it!

File: src/govuk/macros/i18n.njk (hunk 1 above)
[the remark this answers is not in the document]
Comment (romaricpascal) [reply src/govuk/macros/i18n.njk:12]: Couldn't find anything regarding standard ways. My personal preference would lean towards:
- for a single paragraph, it's fine for the `{#`/`#}` to be on the same line
- for multiple paragraphs, it's more readable if the `{#`/`#}` are on separate lines

[the remark this answers is not in the document]
Comment (colinrotherham) [reply src/govuk/macros/i18n.njk:12]: @romaricpascal With the 80 character wrapping that we tend to use for JSDoc too?

Just need to narrow down the indent level. You prefer Flavour 2 (see [comment](https://github.com/alphagov/govuk-frontend/pull/3197#discussion_r1085034742)) still?

### Flavour 1
```njk
{# As the cookie banner component does not currently include JavaScript, we
  have to take care of the 'Hide' button to make progressive enhancement in
  the example functional #}
```

### Flavour 2
```njk
{#
As the cookie banner component does not currently include JavaScript, we have
to take care of the 'Hide' button to make progressive enhancement in the
example functional
#}
```

### Flavour 3
```njk
{#
  As the cookie banner component does not currently include JavaScript, we have
  to take care of the 'Hide' button to make progressive enhancement in the
  example functional
#}
```

[the remark this answers is not in the document]
Comment (romaricpascal) [reply src/govuk/macros/i18n.njk:12]: Honestly, I'm easy as long as it's readable, given there's no strong conventions nor anything to autoformat (if you know of a magic tool somewhere 😉 ). 

I was just thinking Flavour 1 without the indent for single paragraph stuff and Flavour 2 for multi-paragraph as they seem the least cumbersome to type:
- you just carry on typing your paragraph
- you don't worry about any indentation with multiple paragraphs, you just hit enter. 

If you have stronger preferences or one of our tools fixes things in a specific way, happy to align with that and get this through the finish line 🏁 .

**Comment by colinrotherham** [comment]
Rebased with `main` so we can discuss Prettier more without it being a blocker:

* https://github.com/alphagov/govuk-frontend/pull/3201

**Comment by colinrotherham** [comment]
Some of our `.editorconfig` changes will benefit wider GDS too so I've opened:

* https://github.com/alphagov/gds-way/pull/789

**Review by 36degrees — APPROVED** [review]

**Pull request merged**